### ISSUE TITLE
最新修改的代码好像还是有一定的问题

### ISSUE THREAD

**Issue by kuyedie520** [chenhao2345/ABMT#4]
print ('Test on the best model.')
    checkpoint = load_checkpoint(osp.join(args.logs_dir, 'model_best.pth.tar'))
    model_best = models.create(args.arch, num_features=args.features, dropout=args.dropout, num_classes=checkpoint['num_ids'])
    model_best.cuda()
    model_best = nn.DataParallel(model_best)
    evaluator_best = Evaluator(model_best)
    model_best.load_state_dict(checkpoint['state_dict'])
    evaluator_best.evaluate(test_loader_target, dataset_target.query, dataset_target.gallery, cmc_flag=True)
最新修改的是这样的，但是源代码中是用的平均网络去评估的，你这个测试的时候是用的保存的最优模型，让其num_class等于其最优模型的聚类数目。那你现在修改的代码是用的下分支网络评估的还是平均网络评估的，论文中说的好像是平均网络评估的

**Comment by chenhao2345** [comment]
修改后还是用平均网络model_ema评估的，因为model_best.pth.tar里存的是平均网络model_ema的参数。

**Comment by kuyedie520** [comment]
非常感谢您的回复，已经解决
